File: app/src/main/java/mishka/mingeo/di/component/ActivityComponent.java
package mishka.mingeo.di.component;

import android.support.v4.app.FragmentActivity;
import android.support.v4.app.FragmentManager;

import dagger.Component;
import mishka.mingeo.di.ActivityScope;
import mishka.mingeo.di.module.ActivityModule;

@Component(modules = {ActivityModule.class})
public interface ActivityComponent {
    FragmentActivity getActivity();
    FragmentManager getFragmentManager();
}

File: app/src/main/java/mishka/mingeo/di/component/ActivityComponent.kt
package mishka.mingeo.di.component

import android.support.v7.app.AppCompatActivity

import dagger.Component
import mishka.mingeo.data.datamanager.DataManager
import mishka.mingeo.di.BaseActivityScope
import mishka.mingeo.di.module.ActivityModule
import mishka.mingeo.di.module.BoreholeActivityModule
import mishka.mingeo.di.module.PumpingListModule
import mishka.mingeo.di.module.PumpingModule

@Component(dependencies = [ApplicationComponent::class], modules = [ActivityModule::class, BoreholeActivityModule::class, PumpingModule::class, PumpingListModule::class])
@BaseActivityScope
interface ActivityComponent {
    fun getDataManager(): DataManager
    fun getActivity(): AppCompatActivity
}
